_posts/2021/2021-11-29-tidy-up-your-routes-with-only.md

```diff
@@ -0,0 +1,53 @@
+---
+title: "Tidy Up Your Routes with Only"
+description: "Be tidy"
+layout: article
+category: ruby
+image:
+  base: "2021/tidy-up-your-routes-with-only"
+  alt: "Road Closed. Diversion. Road signs."
+  credit: "Ben Wicks"
+  source: "https://unsplash.com/photos/zwN1MwCtR5Y"
+
+---
+
+It is easy with [Rails’ syntax for defining routes](https://guides.rubyonrails.org/routing.html) to open up more URLs in your application than you might intend.
+
+
+## Instead of…
+
+…using an open `resources` block in your routes:
+
+```ruby
+resources :orders do
+  resources :products
+end
+```
+
+## Use…
+
+…the `only` and `except` options to limit the actions you’re generating:
+
+```ruby
+resources :orders, except: %w[destroy] do
+  resources :products, only: %w[show]
+end
+```
+
+
+## Why?
+
+This is a case where it is all about clarity, tidying up and pretecting against unexpected errors or security holes.
+
+Primarily the issue is that in the default non-restricted case the seven default routes are created. If the correct path and method are passed into your application by a user, Rails will atempt to call the relevent controller action, even if it doesn’t exist, which will lead to errors.
+
+There’s also the chance of security holes. Rails’ standardised naming (while great for our code organisation and structure) means routes can be guessed. The presence of a `GET /orders/12345` route pointing to `orders#show` would imply the existance of a `GET /orders` route for `orders#index` and more dangerously a `PUT /orders/12345` route for `orders#update`.
```

Comment:
With the first paragraph of this sentence (and the first sentence of the paragraph before), I wonder if the first paragraph of this section is needed! I say not!

* Rails's (I don't like it but it's grammatically correct to include the 's' if not talking about a pluralised thing)

* comma needed before 'and more dangerously...'